sdk/core/azure-core/src/main/java/com/azure/core/http/policy/MaxAttemptRedirectStrategy.java

```diff
@@ -0,0 +1,91 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.azure.core.http.policy;
+
+import com.azure.core.http.HttpMethod;
+import com.azure.core.util.logging.ClientLogger;
+
+import java.util.HashSet;
+import java.util.Objects;
+import java.util.Set;
+
+/**
+ * A default implementation of {@link RedirectStrategy} that uses the provided try count, and status code
+ * to determine if request should be redirected.
+ */
+public class MaxAttemptRedirectStrategy implements RedirectStrategy {
+
+    // Based on Stamp specific redirects design doc
+    static final int MAX_REDIRECT_ATTEMPTS = 10;
+    private static final String LOCATION_HEADER_NAME = "Location";
+
+    private int maxAttempts;
+    private final String locationHeader;
```

Comment:
Thoughts on having this be a `Function<HttpHeaders, String> redirectLocationRetriever`? This would allow for more flexibility, unless the plan is that another implementation of `RedirectStrategy` would be used.